src/ice/agent.rs

```diff
@@ -2178,4 +2195,36 @@ mod test {
 
         buf
     }
+
+    #[test]
+    fn relayed_candidates_across_ip_versions_have_lower_priority() {
+        let mut agent = IceAgent::new();
+
+        let relay_ipv4_ipv4 = relay("1.1.1.1:5000", "2.2.2.2:5000", "udp");
+        let relay_ipv4_ipv6 = relay("1.1.1.1:5001", "[::1]:5000", "udp");
+        let relay_ipv6_ipv4 = relay("[::1]:5000", "1.1.1.1:5000", "udp");
+        let relay_ipv6_ipv6 = relay("[::1]:5001", "[::2]:5000", "udp");
+
+        agent.add_local_candidate(relay_ipv4_ipv4.clone());
+        agent.add_local_candidate(relay_ipv6_ipv6.clone());
```

Comment:
Something worth noting is that the order of these candidates here is important to make the test pass due to priority ordering performed in `add_local_candidate`. So we need to add this IPv6 candidate first before the other one to ensure this priority is higher than the IPv4 <> IPv4 one.

I couldn't find a way of working around that.

Reply:
I just tried to add them in another order and it still works.

Reply:
Interesting, why does this suddenly work? I need to check again.